### DISCUSSION TITLE
best practice for document migration/upgrade

### DISCUSSION THREAD

**Discussion by hshenMcc** [ravendb/ravendb#15903] [Q&A] [answered]
Current Payment collection has a few million documents recoding Visa credit card only with class as
public class Payment {
    public string Id {get ; set;}
    public string CustomerName {get; set;}
    public string VisaNum{get; set;}
    public decimal Amount {get; set;}
    public string HolderName {get; set;}
}
Now the app will process both Visa and Master, with classes
public class CreditCard
{
   public string CardType {get; set;}
    public string CardNum{get; set;}
    public string HolderName {get; set;}
}
public class Payment {
    public string Id {get ; set;}
    public string CustomerName {get; set;}
    public decimal Amount {get; set;}
    public CreditCard Card{get; set;}
}
How may I migrate the existing documents to match the new classes as
     Card.CardType = "Visa"
     Card.CardNum = VisaNum
     Card.HolderName = HolderName
Can this be done through a scripting similar as
    insert into newTable
    select 
    from oldTable
in SQL world?

Thanks.

**Comment by georgiosd** [comment, marked as answer]
This works quite well: https://github.com/olibanjoli/RavenDb.Migrations

**Reply by hshenMcc** [reply]
Thanks. 

I tried the tool. It deals with migrating multiple collections all in one run. I only need to migrate/upgrade **one** class/collection. I want to know how to efficiently do that? Basically, how do I implement the Up() method? 

I don't think it's a good idea to load all the documents from Ravendb to your local machine, upgrade them and save back. In SQL world, "insert select" will avoid around trip from server to client and back. I would like to know if there is an equivalent operation in Ravendb. May it not be feasible to achieve in Ravendb because schema less nature?

**Reply by georgiosd** [reply]
Why do you think it "loads all documents from RavenDB to your local machine"? It doesn't.

Use `PatchCollection` method and write your set patch. 

If you don't want the package, you can it yourself like so: https://ravendb.net/docs/article-page/5.4/csharp/client-api/operations/patching/set-based

Or manually from studio: https://ravendb.net/docs/article-page/5.4/csharp/studio/database/documents/patch-view

**Reply by hshenMcc** [reply]
Thank you very much. Patch View in Ravendb Studio is what I am looking for.
